frame/support/src/weights.rs

```diff
@@ -236,6 +247,35 @@ impl SimpleDispatchInfo {
 	}
 }
 
+/// A struct to represent a weight which is a function of the input arguments. The given closure's
+/// arguments must match the dispatch's argument types, given as a tuple.
+pub struct FunctionOf<F, Class>(pub F, pub Class, pub bool);
+
+impl<Args, F, Class> WeighData<Args> for FunctionOf<F, Class>
+where
+	F : Fn(Args) -> Weight
+{
+	fn weigh_data(&self, args: Args) -> Weight {
+		(self.0)(args)
+	}
+}
+
+impl<Args, F, Class> ClassifyDispatch<Args> for FunctionOf<F, Class>
+where
+	Class: Into<DispatchClass> + Clone,
```

Comment:
I'm not entirely sure about the interest of having the Class abstract here ? can't we just have DispatchClass as second field of FunctionOf directly ?

Reply:
Yeah I will change it to that. The purpose was to provide a syntactic simplicity. I will do that later with a macro.

Reply:
removed.